app/assets/javascripts/components/activity/recent_uploads_handler.jsx

```diff
@@ -2,16 +2,41 @@ import React from 'react';
 import createReactClass from 'create-react-class';
 import PropTypes from 'prop-types';
 import { connect } from "react-redux";
-import UploadTable from './upload_table.jsx';
 import { fetchRecentUploads, sortRecentUploads } from '../../actions/recent_uploads_actions.js';
+import Loading from '../common/loading.jsx';
+import Upload from '../uploads/upload.jsx';
 
-const HEADERS = [
-      { title: I18n.t('recent_activity.image'), key: 'image' },
-      { title: I18n.t('recent_activity.file_name'), key: 'file_name' },
-      { title: I18n.t('recent_activity.uploaded_by'), key: 'username', style: { minWidth: 160 } },
-      { title: I18n.t('recent_activity.usage_count'), key: 'usage_count', style: { width: 160 } },
-      { title: I18n.t('recent_activity.datetime'), key: 'uploaded_at', style: { width: 200 } },
-    ];
+
+export const UploadTable = createReactClass({
```

Comment:
This component should have its own file.

Reply:
So bring back the upload_table? 

I saw there is another `export const RecentUploadsHandlerBase` in the same file. It's legal because it's part of the `export default connect` while `UploadTable` is not?

Also, there is a prop `onSort={this.props.sortRecentUploads}` in the returned Upload table. This no sorting function right now. Is it for later? Should I keep it?

Reply:
`RecentUploadsHandlerBase` was the only component defined in the file; while `export default connect()` returns a component, it doesn't call createReactClass or similar, so it doesn't trigger the eslint rule.

I missed this before, but the last change didn't fix the duplication issue. We should keep either UploadTable or UploadList — probably UploadList, for consistency with the similar ArticleList and RevisionList components — and use that component in both UploadHandler and RecentUploadsHandler.

The sorting function is working currently for the recent uploads tab: https://dashboard.wikiedu.org/recent-activity/recent-uploads

upload_actions.js has a similar sorting function that gets connected in UploadsHandler. With gallery design, there is no sorting to be done via table headers, so we'll need to add a sorting `select` that uses the sorting function in RecentUploadsHandler, like in UploadsHandler.